tests/features/fixtures/mazerunner/src/main/cpp/bugsnags.cpp

```diff
@@ -291,4 +291,82 @@ Java_com_bugsnag_android_mazerunner_scenarios_CXXConfigurationMetadataNativeCras
   return 12167;
 }
 
+JNIEXPORT void JNICALL
+Java_com_bugsnag_android_mazerunner_scenarios_CXXStartScenario_activate(
+    JNIEnv *env,
+    jobject instance) {
+  bugsnag_start(env);
+  bugsnag_leave_breadcrumb((char *)"Start scenario crumb", BSG_CRUMB_LOG);
+  bugsnag_notify((char *)"Start scenario",
+                     (char *)"Testing env", BSG_SEVERITY_INFO);
+}
+
+JNIEXPORT void JNICALL
+Java_com_bugsnag_android_mazerunner_scenarios_CXXRemoveOnErrorScenario_activate(
+    JNIEnv *env,
+    jobject instance) {
+  bugsnag_add_on_error(&on_err_false);
```

Comment:
I'm not convinced this is going to prove behaviour as intended.... The first callback returning false would mean no further ones are run anyway, right?  So I don't think adding and then removing a true return would prove what we want to.

Assuming I've understood everything correctly, it might be better to create a second true-returning callback that set different fields to the first.  Something along the lines of:
1. Add the first callback
2. (Optionally) notify 
3. add the second
4. notify
5. remove callback
6. notify one more time.... 

Where we'd expect 3 events to be sent, with the 1st and 3rd being the same and the 2nd having addition fields set.

Reply:
I've actually checked the implementation and I don't think this is necessary for proving the `on_error` callbacks can be removed.  It seems there can only be one callback present in the NDK layer at once, so to verify this we'd only need to:
1. Set some data
2. Add a callback that mutates said data
2. Notify
3. Verify the data was mutated
4. Remove the callback
5. Notify
6. Verify the data is not mutated

Reply:
Ok, I like that approach too.